### PR TITLE
Abid dan/codegen

### PR DESCRIPTION
**Pull request by aramay**
refactor codeGen factory completed.

### CODE REVIEW

File: helpers/codeGen/parseCSSText.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const parseCSSText = (cssText) => {
```
Comment (theTschida) [inline helpers/codeGen/parseCSSText.js:1]: Please update 'var' to ES6 standards.

File: pages/api/genearteReactCodeString.js [hunk 2]
```diff
@@ -1,182 +1,15 @@
+import { htmlGenerator } from '../../helpers/codeGen/htmlGenerator';
+
 export default function genearteReactCodeString(req, res) {
   const prettier = require('prettier');
   try {
-    const { layout, components } = req.body;
-    console.log('layout', layout)
+    const { layout, components, framework } = req.body;
     let result = [];
```
Comment (theTschida) [inline pages/api/genearteReactCodeString.js:7]: Use const with objects  and arrays.

File: pages/api/genearteReactCodeString.js [hunk 3]
```diff
@@ -1,182 +1,15 @@
...
-
-    const eleFactory = new HTMLElementFactory();
-
+    
     const parseComponents = (coms, components) => {
       for (const component of coms) {
-        const fullComponentDetails = components.find(c => c.id === component.id)
-        console.log('fullComponentDetails', fullComponentDetails)
-        if (fullComponentDetails) result.push(eleFactory.createElement(fullComponentDetails).div); 
+        const fullComponentDetails = components.find((c) => c.id === component.id);
```
Comment (theTschida) [inline pages/api/genearteReactCodeString.js:11]: What is 'c'?

File: pages/api/genearteReactCodeString.js [hunk 4]
```diff
@@ -1,182 +1,15 @@
...
+        if (fullComponentDetails) result.push(htmlGenerator(fullComponentDetails, framework))
```
Comment (theTschida) [inline pages/api/genearteReactCodeString.js:12]: This style is inconsistent. In this example, the code is all on one line. On line 19, it is in curly braces and on a separate line.

**Review by theTschida — COMMENTED** [review]
Please add comments to the code. Explaining the "why" of a function is very helpful.

File: pages/api/genearteReactCodeString.js (hunk 3 above)
Comment (bamche) [reply pages/api/genearteReactCodeString.js:11]: emily's comments are making this hard to read

**Review by bamche — CHANGES_REQUESTED** [review]
great stuff, can you walk us through resolving some of these comments?

**Pull request merged**